src/cultionet/data/create.py

```diff
@@ -567,69 +582,76 @@ def create_predict_dataset(
 
-            with tqdm(
-                total=len(windows),
-                desc='Creating prediction windows',
-                position=0
-            ) as pbar_total:
-                with parallel_backend(
-                    backend='loky',
-                    n_jobs=num_workers
-                ):
-                    for window_chunk in get_window_chunk(windows, chunksize):
-                        with TqdmParallel(
-                            tqdm_kwargs={
-                                'total': len(window_chunk),
-                                'desc': 'Window chunks',
-                                'position': 1,
-                                'leave': False
-                            },
-                            temp_folder='/tmp'
-                        ) as pool:
-                            __ = pool(
-                                delayed(partial_create)(
-                                    read_slice(time_series, window_pad), window, window_pad
-                                ) for window, window_pad in window_chunk
-                            )
-                        pbar_total.update(len(window_chunk))
+                partial_create = partial(
+                    create_and_save_window,
+                    process_path,
+                    ntime,
+                    nbands,
+                    src_ts.gw.nrows,
+                    src_ts.gw.ncols,
+                    ref_res,
+                    resampling,
+                    region,
+                    year,
+                    window_size,
+                    padding
+                )
+
+                with tqdm(
+                    total=len(windows),
+                    desc='Creating prediction windows',
+                    position=1
+                ) as pbar_total:
+                    with parallel_backend(
+                        backend='loky',
+                        n_jobs=num_workers
+                    ):
+                        for window_chunk in get_window_chunk(windows, chunksize):
+                            with TqdmParallel(
+                                tqdm_kwargs={
+                                    'total': len(window_chunk),
+                                    'desc': 'Window chunks',
+                                    'position': 2,
+                                    'leave': False
+                                },
+                                temp_folder='/tmp'
+                            ) as pool:
+                                __ = pool(
+                                    delayed(partial_create)(
+                                        read_slice(time_series, window_pad),
+                                        window,
+                                        window_pad
+                                    ) for window, window_pad in window_chunk
+                                )
+                            pbar_total.update(len(window_chunk))
 
 
 def create_dataset(
```

Comment:
This thing is getting very long. Probably a good idea to find logical chunks to wrap in functions. See here for some guidelines on how to tell when functions are getting too long: https://stackoverflow.com/questions/475675/when-is-a-function-too-long.